src/structures/ReactionCollector.js

```diff
@@ -0,0 +1,153 @@
+const EventEmitter = require('events').EventEmitter;
+const Collection = require('../util/Collection');
+
+/**
+ * Collects reactions on a message.
+ * @extends {EventEmitter}
+ */
+class ReactionCollector extends EventEmitter {
```

Comment:
Since ReactionCollector is very similar to MessageCollector, it would be great if you created a Collector interface class that both are able to use to minimize code duplication.